opensquirrel/merger.py

```diff
@@ -84,3 +85,6 @@ def merge_single_qubit_gates(squirrel_ir: SquirrelIR):
     for accumulated_bloch_sphere_rotation in accumulators_per_qubit.values():
         if not accumulated_bloch_sphere_rotation.is_identity():
             squirrel_ir.statements.append(accumulated_bloch_sphere_rotation)
+
+    squirrel_ir.statements = sorted(squirrel_ir.statements, key=lambda obj: isinstance(obj, Measure))
```

Comment:
What does this do? Does it do a resort such that the Measurements are placed at the end? Is the order of the other statements preserved (i.e. how they relate two any 2 qubit statements appearing)? Is this tested sufficiently?

Reply:
Yes, this is due to the decomposition step which puts them in the middle. I will create more tests.